stk/calculators/energy/energy_calculators.py

```diff
@@ -377,3 +382,887 @@ def energy(self, mol, conformer=-1):
+
+    See Also
+    --------
+    #. https://xtb-docs.readthedocs.io/en/latest/setup.html
+
+    """
+    def __init__(self,
+                 xtb_path,
+                 gfn_version='2',
+                 output_dir=None,
+                 num_cores=1,
+                 electronic_temperature=300,
+                 solvent=None,
+                 solvent_grid='normal',
+                 charge=0,
+                 unpaired_electrons=0,
+                 use_cache=False,
+                 unlimited_memory=False):
+        """
+        Initializes a :class:`XTBEnergy` instance.
+
+        Parameters
+        ----------
+        xtb_path : :class:`str`
+            The path to the xTB executable.
+
+        gfn_version : :class:`str`, optional
+            Parameterization of GFN to use in xTB.
+            For details see
+            https://xtb-docs.readthedocs.io/en/latest/basics.html
+
+        output_dir : :class:`str`, optional
+            The name of the directory into which files generated during
+            the optimization are written, if ``None`` then
+            :func:`uuid.uuid4` is used.
+
+        num_cores : :class:`int`, optional
+            The number of cores xTB should use.
+
+        electronic_temperature : :class:`int`, optional
+            Electronic temperature to use (in K).
+
+        solvent : :class:`str`, optional
+            Solvent to use in GBSA implicit solvation method.
+            For details see
+            https://xtb-docs.readthedocs.io/en/latest/gbsa.html
+
+        solvent_grid : :class:`str`, optional
+            Grid level to use in SASA calculations for GBSA implicit
+            solvent.
+            Can be one of ``'normal'``, ``'tight'``, ``'verytight'``
+            or ``'extreme'``
+            For details see
+            https://xtb-docs.readthedocs.io/en/latest/gbsa.html
+
+        charge : :class:`int`, optional
+            Formal molecular charge.
+
+        unpaired_electrons : :class:`int`, optional
+            Number of unpaired electrons.
+
+        use_cache : :class:`bool`, optional
+            If ``True`` :meth:`energy` will not run twice on the same
+            molecule and conformer.
+
+        unlimited_memory : :class: `bool`, optional
+            If ``True`` :meth:`energy` will be run without constraints
+            on the stacksize. If memory issues are encountered, this
+            should be ``True``, however this may raise issues on
+            clusters.
+
+        """
+        self.xtb_path = xtb_path
+        self.gfn_version = gfn_version
+        self.output_dir = output_dir
+        self.num_cores = str(num_cores)
+        self.electronic_temperature = str(electronic_temperature)
+        self.solvent = solvent
+        if self.solvent is not None:
+            self.solvent = solvent.lower()
+            valid_xtb_solvent(
+                gfn_version=self.gfn_version,
+                solvent=self.solvent
+            )
```

Comment:
Normally the check that the input parameters are correct would be done before the state of the isntance is modified. Ie, before you do 

```python

self.xtb_path = xtb_path

```

You would have 

```python

if solvent is not None:
    solvent = solvent.lower()
    if not is_valid_solvent(gfn_version, solvent):
        raise XTBInvalidSolventError(
            f'Solvent {solvent!r} is invalid for '
            f'version {gfn_version!r}.'
        )

self.xtb_path = xtb_path
#stuff 
self.solvent = solvent

```

Note the `{solvent!r}` and `{gfn_version!r}` change to show the types in the error message.
Also note that the function now returns True or False rather than raising. The raising is done in the place where the error occurs and the function behaves more consistently. Ie returning True or False makes more sense than only returning True and raising instead of returning False, in my opinion anyway. I also think that putting `is_` in front of the function reads better and also makes it more clear what the function does, ie it makes it pretty clear that the function performs a a test which returns a bool.

Reply:
Very good points @lukasturcani